SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialRPCService.cpp

```diff
@@ -36,8 +36,13 @@ EPushRPCResult SpatialRPCService::PushRPC(Worker_EntityId EntityId, ERPCType Typ
 		TOptional<Trace_SpanId> CauseSpanId = EventTracer->GetFromStack();
 		TOptional<Trace_SpanId> SpanId =
 			CauseSpanId.IsSet() ? EventTracer->CreateSpan(&CauseSpanId.GetValue(), 1) : EventTracer->CreateSpan();
-		EventTracer->TraceEvent(FSpatialTraceEventBuilder::CreateSendRPC(Target, Function), SpanId);
+
+		EventTraceUniqueId TraceId = SpanId.IsSet() ? EventTraceUniqueId::GenerateUnique() : EventTraceUniqueId{};
```

Comment:
I think we always want to do EventTraceUniqueId::GenerateUnique() here? The existing CauseSpanId.IsSet() check to set the spanId is just to work out if the send rpc was caused by user event.

Reply:
Yeah true, should this whole block be inside EventTracer->IsEnabled()?

Reply:
Yeah kind of. It should all work if event tracing is disabled but... maybe slightly more efficient to check once rather than many time for enabled.